docs/rest.md

```diff
@@ -0,0 +1,623 @@
+     "name": "dd",
+     "origin": "canonical",
+     "resource": "/1.0/packages/dd.canonical",
+     "status": "not installed",
+     "type": "app",
+     "vendor": "Somebody",
+     "version": "0.1"
+   },
+   "pastebinit.mvo": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "http://storeurl/icon.png",
+     "name": "pastebinit.mvo",
+     "installed_size": "-1",
+     "operation": {
+       "created_at": 1415639996,
+       "may_cancel": true,
+       "resource": "/1.0/packages/pastebinit.mvo",
+       "status": "Running",
+       "status_code": 100,
+       "updated_at": 1415639996
+     },
+     "origin": "mvo",
+     "resource": "/1.0/packages/pastebinit.mvo",
+     "rollback_available": "0.9",
+     "status": "installing",
+     "type": "app",
+     "update_available": "1.1",
+     "vendor": "Michael Vogt",
+     "version": "0.8"
+   },
+   "ubuntu-core.canonical": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "",               // core might not have an icon
+     "installed_size": "-1",   // core doesn't have installed_size (yet)
+     "name": "ubuntu-core",
+     "origin": "canonical",
+     "resource": "/1.0/packages/ubuntu-core.canonical",
+     "status": "active",
+     "type": "core",
+     "vendor": "Canonical",
+     "version": "43"
+   }
+ }
+}
+```
+
+#### Fields
+* `packages`
+    * `status`: can be either `not installed`, `installing`, `installed`,
+      `uninstalling`, `active` (i.e. is current), `removed` (but data
+      present), `purging`; there is no `purged` state, as a purged package is
+      undistinguishable from a non-installed package. For statuses that signal
+      a background operation is in course, see the `operation`
+      field. Transient states not implemented yet.
+    * `name`: the package name.
+    * `version`: a string representing the version.
+    * `vendor`: a string representing the vendor.
+    * `icon`: a url to the package icon, possibly relative to this server.
+    * `type`: the type of snappy package; one of `app`, `framework`, `kernel`,
+      `gadget`, or `os`.
+    * `description`: package description
+    * `installed_size`: for installed packages, how much space the package
+      itself (not its data) uses, formatted as a decimal string.
+    * `download_size`: for not-installed packages, how big the download will
+      be, formatted as a decimal string.
+    * `operation`: if the state signals that an operation is underway
+      (e.g. installing), the operation field describes that operation
+    * `rollback_available`: if present and not empty, it means the package can
+      be rolled back to the version specified as a value to this entry.
+    * `update_available`: if present and not empty, it means the package can be
+      updated to the version specified as a value to this entry.
+
+### POST
+
+* Description: Sideload a package to the system. 
```

Comment:
About config, I agree, it's weird (and not clear whether it's needed, as you can already configure individual pacakges elsewhere).

About POST I disagree, as POSTing to the collection URL is the way to add a new thing to the collection.